src/ui/EditTask.svelte

```diff
@@ -35,59 +36,53 @@
     let parsedRecurrence: string = '';
     let parsedDone: string = '';
 
-    $: {
-        if (!editableTask.startDate) {
-            parsedStartDate = '<i>no start date</>';
-        } else {
-            const parsed = chrono.parseDate(
-                editableTask.startDate,
-                new Date(),
-                {
-                    forwardDate: true,
-                },
-            );
-            if (parsed !== null) {
-                parsedStartDate = window.moment(parsed).format('YYYY-MM-DD');
-            } else {
-                parsedStartDate = '<i>invalid start date</i>';
-            }
+    function doAutocomplete(date: string): string {
+        for (let [key, val] of Object.entries(Abbrev)) {
+            date = date.replace(RegExp(`\\b${key}\\s`), val);
         }
+        return date;
     }
 
-    $: {
-        if (!editableTask.scheduledDate) {
-            parsedScheduledDate = '<i>no scheduled date</>';
-        } else {
-            const parsed = chrono.parseDate(
-                editableTask.scheduledDate,
-                new Date(),
-                {
-                    forwardDate: true,
-                },
-            );
-            if (parsed !== null) {
-                parsedScheduledDate = window
-                    .moment(parsed)
-                    .format('YYYY-MM-DD');
-            } else {
-                parsedScheduledDate = '<i>invalid scheduled date</i>';
-            }
+    function parseDate(
+        type: 'start' | 'scheduled' | 'due' | 'done',
```

Comment:
Thank you very much for refactoring this to remove the repetition. Much appreciated!